src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -535,51 +532,47 @@ public with sharing class PMT_PaymentWizard_CTRL {
     }
 
     /*******************************************************************************************************
-    * @description Displays an error message when then user has no permission to modify payments
-    * @return void
+    * @description Returns true if the user has permissions to access, create and delete npe01__oppPayment__c
+    * and all relevant fields are both readable and modifiable
+    * @return Boolean
     */
-    private void displayNoAccessMessage() {
-        String errorMsg;
+    private Boolean hasFeatureAccess() {
+        SObjectType payment = npe01__OppPayment__c.getSObjectType();
+        SObjectType opportunity = Opportunity.getSObjectType();
+        return hasAccessTo(payment, paymentFields())
+            && hasAccessTo(opportunity, opportunityFields());
+    }
 
-        if (isWriteoffWizard == true) {
-            errorMsg = System.Label.pmtWizardWriteoffNoAccessMessage;
-        } else if (isScheduleWizard == true) {
-            errorMsg = System.Label.pmtWizardPaymentsNoAccessMessage;
-        } else {
-            errorMsg = System.Label.pmtWizardMsgNoOppFound;
+    private Boolean hasAccessTo(SObjectType sObjectType, Set<SObjectField> sObjectFields) {
+        if (canCheckPermissionsFor(sObjectType)) {
+            return permissions.canRead(sObjectType, sObjectFields)
+                && permissions.canUpdate(sObjectType, sObjectFields);
         }
+        return false;
+    }
 
-        ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, errorMsg));
+    private Boolean canCheckPermissionsFor(SObjectType objectType) {
```

Comment:
Can we remove this method if it's no longer being used?

Reply:
`canCheckPermissionsFor` is still being used in `hasAccessTo`. That check is a gated boolean, instead of cluttering the `if` statement with ` permissions.canCreate(objectType) && permissions.canDelete(objectType);`. I think it is better to have it in its own private method.

Reply:
Got ya. Missed that and Intellij didn't find usages for some reason.